sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -554,12 +556,14 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
     }
 
     val command = (mode, tableOpt) match {
-      case (_, Some(table: V1Table)) =>
+      case (_, Some(_: V1Table)) =>
         return saveAsTable(TableIdentifier(ident.name(), ident.namespace().headOption))
 
       case (SaveMode.Append, Some(table)) =>
         checkPartitioningMatchesV2Table(table)
-        AppendData.byName(DataSourceV2Relation.create(table), df.logicalPlan, extraOptions.toMap)
+        val v2Relation =
+          DataSourceV2Relation.create(table, catalogManager.catalogIdentifier(catalog), Seq(ident))
+        AppendData.byName(v2Relation, df.logicalPlan, extraOptions.toMap)
 
       case (SaveMode.Overwrite, _) =>
```

Comment:
A little curious why `Overwrite` doesn't need to create a `DataSourceV2Relation`?

Reply:
It drops and recreates a table. It's a DDL operation instead of DML

Reply:
Still probably a dumb question. Why does DDL/DML affects how we generate the query plan? I'm asking this because in the `save()` function for the DataFrameWriter, we do generate a `DataSourceV2Relation` for `Overwrite ` mode. I'm curious about why there is such a difference here.

Reply:
in save, we don't update table information in a catalog. saveAsTable updates the catalog, therefore is doing different work. So we need to do a separate operation